[GEOT-7767] Java UUID type support to geopkg dialect

tuomari opened the pull request:
[![GEOT-7767](https://badgen.net/badge/JIRA/GEOT-7767/0052CC)](https://osgeo-org.atlassian.net/browse/GEOT-7767) [<img width="16" alt="Powered by Pull Request Badge" src="https://user-images.githubusercontent.com/1393946/111216524-d2bb8e00-85d4-11eb-821b-ed4c00989c02.png">](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)<!-- PR-BADGE: PLEASE DO NOT REMOVE THIS COMMENT -->

When generating geopkg with java.util.UUID type, for example from geoserver, the export fails with exception: 
```
No mapping for java.util.UUID
java.io.IOException: Error occurred creating table
        at org.geotools.jdbc.JDBCDataStore.createSchema(JDBCDataStore.java:743)
        at org.geotools.geopkg.GeoPackage.create(GeoPackage.java:824)
        at org.geotools.geopkg.GeoPackage.add(GeoPackage.java:882)
  ....
  ```
  
  This PR will add mapping of java.util.UUID types to VARCHAR

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

tuomari commented:
I found an old jira ticket regarding the same problem: https://osgeo-org.atlassian.net/browse/GEOS-10728

jurgispods commented:
We need the very same feature and would be glad to see this getting merged.

tuomari commented:
@mprins ticket has been created, and commit message updated. I don't believe there is need for documentation changes, and all other items in the checklist are checked. Is there anything else we need to do before this MR can be merged?

aaime commented:
The downstream build is failing because we switched from using JAItools zonal stats to JAIExt ones. I guess this PR is based on a version of Geotools that's older than the switch. Could you rebase it to the current main branch?

tuomari commented:
> The downstream build is failing because we switched from using JAItools zonal stats to JAIExt ones. I guess this PR is based on a version of Geotools that's older than the switch. Could you rebase it to the current main branch?

Rebased on main
